tests/test_api.rs

```diff
@@ -4515,3 +4515,47 @@ fn run_with_rust_allocator() {
   let count_loaded = count.load(Ordering::SeqCst);
   assert_eq!(count_loaded, 0);
 }
+
+#[test]
+fn oom_callback() {
+  extern "C" {
+    fn setjmp(env: &mut [usize; 32]) -> i32;
+    fn longjmp(env: &[usize; 32], value: i32) -> !;
+  }
+
+  static mut JMPBUF: [usize; 32] = [0; 32];
+
+  extern "C" fn oom_callback(_: *const std::os::raw::c_char, _: bool) {
+    unsafe {
+      longjmp(&JMPBUF, 42);
```

Comment:
This isn't safe (and indeed the test suite is failing on the windows buildbot) because V8 - possibly the Rust runtime too - is in no way longjmp-proof.

V8 heavily depends on RAII. When you longjmp, no destructors run and you basically clobber its internal state. That affects other tests too because they run in the same process.

Basically, you can test up to the `isolate.set_oom_error_handler(oom_callback)` call, but no further.

Reply:
Yeah it seems that the test is broken on Windows, triggering some undefined behavior here.

Updated.